foreshadow/core/fs_logging.py

```diff
@@ -0,0 +1,264 @@
+"""Logging methods interfacing with a potential GUI system."""
+import atexit as _atexit
+import logging as _logging
+import sys as _sys
+import threading as _threading
+
+
+_logger = None
+_logger_lock = _threading.Lock()
+
+_this_module = _sys.modules[__name__]
+
+LEVELS = {
+    "debug": _logging.DEBUG,
+    "info": _logging.INFO,
+    "warning": _logging.WARNING,
+    "error": _logging.ERROR,
+    "critical": _logging.CRITICAL,
+}
+
+
+def get_logger():
+    """Return Foreshadow logger instance.
+
+    Will create and setup if needed, else will return the previously setup
+    logger.
+
+    Returns:
+        foreshadow logger instance.
+
+    """
+    global _logger
+
+    # if already created, don't acquire lock
+    if _logger:
+        return _logger
+
+    _logger_lock.acquire()
+
+    try:
+        if _logger is not None:
+            return _logger
+
+        # Get scoped Foreshadow logger.
+        _my_logger = _logging.getLogger("foreshadow")
```

Comment:
to prevent users from interacting with it outside of the determined methods, which could screw around with the behaviour of it, I think (I trusted tensorflow on this one)

Reply:
will make non-private